src/main/kotlin/com/tschuchort/compiletesting/KotlinCompilation.kt

```diff
@@ -403,6 +410,13 @@ class KotlinCompilation {
 
 	/** Performs the 1st and 2nd compilation step to generate stubs and run annotation processors */
 	private fun stubsAndApt(sourceFiles: List<File>): ExitCode {
+		pluginClasspaths.forEach { filepath ->
+			if (!filepath.exists()) {
+				error("Plugin $filepath not found")
+				return ExitCode.INTERNAL_ERROR
+			}
+		}
+
```

Comment:
I think this should probably be moved to the calling function, where writing of the source files happens as well.

Reply:
Please, which calling function? Thanks in advance!

Reply:
Please, which function? `stubsAndApt` is called from `compile`:

```
            // step 1 and 2: generate stubs and run annotation processors
            try {
                val exitCode = stubsAndApt(sourceFiles)
                if (exitCode != ExitCode.OK) {
                    val messages = internalMessageBuffer.readUtf8()
                    searchSystemOutForKnownErrors(messages)
                    return Result(exitCode, classesDir, messages)
                }
            } finally {
                KaptComponentRegistrar.threadLocalParameters.remove()
            }
```

And previously, the writing of the source files appears:

```
        // write given sources to working directory
        val sourceFiles = sources.map { it.writeIfNeeded(sourcesDir) }
```

Though I don't find an error checking section like the one I found in `stubsAndApt`.

Reply:
I think I would just put it after `val sourceFiles = sources.map { it.writeIfNeeded(sourcesDir) }` in the `compile` function where all the initialization of directories is done. But you are right that it's kinda confusing where to put this stuff. There should be a clear separation between initialization and the different compile stages. Maybe I will rewrite it in the future.